nexus/reconfigurator/planning/src/blueprint_editor/allocators/internal_dns.rs

```diff
@@ -37,9 +47,11 @@ impl DnsSubnetAllocator {
             })
             .collect::<BTreeSet<DnsSubnet>>();
 
-        let redundancy = input.target_internal_dns_zone_count();
-        if redundancy > INTERNAL_DNS_REDUNDANCY {
-            return Err(Error::TooManyDnsServers);
+        // TODO-cleanup This is the only reason we take `target_redundancy`; do
+        // we have another use for it in the future, or should someone else do
+        // this check?
+        if target_redundancy > INTERNAL_DNS_REDUNDANCY {
```

Comment:
This check might be able to move to blippy too? Or maybe the planner? We might be able to drop it altogether - a caller can try to `alloc()` as many subnets as they want, regardless of what they pass here, but we'll fail at allocation time if they try to ask for more than `INTERNAL_DNS_REDUNDANCY` (as we should!).